src/activities/Elsa.Activities.Reflection/Activities/ExecuteFunction.cs

```diff
@@ -0,0 +1,114 @@
+        RuntimeDescription = "a => !!a.state.variableName ? `Execute a Method by reflection and store the result into <strong>${ a.state.variableName }</strong>.` : 'Execute a Method by reflection.'",
+        Outcomes = new[] { OutcomeNames.Done }
+    )]
+    public class ExecuteMethod : Activity
+    {
+
+        public ExecuteMethod(IWorkflowExpressionEvaluator evaluator)
+        {
+
+        }
+
+
+        [ActivityProperty(Hint = "The variables to use as parameters, seperated by comma in order of method call.")]
+        public string InputVariableNames
+        {
+            get => GetState<string>(null, "InputVariableNames");
+            set => SetState(value, "InputVariableNames");
+        }
+
+        [ActivityProperty(Hint = "The name of the variable to store the returned value into.")]
+        public string OutputVariableName
+        {
+            get => GetState<string>(null, "OutputVariableName");
+            set => SetState(value, "OutputVariableName");
+        }
+
+        [ActivityProperty(Hint = "Assembly name (fullname or filename) to load")]
+        public string AssemblyName
+        {
+            get => GetState<string>(null, "AssemblyName");
+            set => SetState(value, "AssemblyName");
+        }
+
+        [ActivityProperty(Hint = "Class name to start or lookup")]
+        public string ClassName
+        {
+            get => GetState<string>(null, "ClassName");
+            set => SetState(value, "ClassName");
+        }
+
+        [ActivityProperty(Hint = "Class is a static class?")]
+        public bool IsStaticClass
+        {
+            get => GetState<bool>(null, "IsStaticClass");
+            set => SetState(value, "IsStaticClass");
+        }
+
+        [ActivityProperty(Hint = "Method name to execute")]
+        public string MethodName
+        {
+            get => GetState<string>(null, "MethodName");
+            set => SetState(value, "MethodName");
+        }
+
+        protected override async Task<ActivityExecutionResult> OnExecuteAsync(
+            WorkflowExecutionContext context,
+            CancellationToken cancellationToken)
+        {
+            await Task.Delay(1);
+            var inputValues = InputVariableNames.Split(',').Select(s => context.GetVariable(s)).ToArray();
+            return Execute(context, inputValues);
+        }
+
+        protected override ActivityExecutionResult OnResume(WorkflowExecutionContext context)
+        {
+            var inputValues = InputVariableNames.Split(',').Select(s => context.GetVariable(s)).ToArray();
+            return Execute(context, inputValues);
+        }
+
+        private ActivityExecutionResult Execute(WorkflowExecutionContext workflowContext, object[] receivedInputValues)
+        {
+            string path = Directory.EnumerateFiles(AppDomain.CurrentDomain.BaseDirectory, AssemblyName, SearchOption.AllDirectories).FirstOrDefault();
+            Assembly assembly = Assembly.LoadFrom(path);
```

Comment:
Why not just use `Type.GetType("Assembly Qualified Type Name")` instead of loading assemblies from disk?

Reply:
If the assembly's not loaded this doesnt work... If i e.g. place three poco dll into the server (without referincing it in the project) they wont get loaded. Should at least load an (optional) binary from specified path to dll. I even dont like the lazy loader anymore since it costs a lot of performance